src/angularComponent/ngcOmniboxController.js

```diff
@@ -1,22 +1,64 @@
+import {KEY, isVerticalMovement} from '../coreComponent/keyboard.js';
+
+// Protects against multiple key events firing in a row without disallowing holding down the key
+const KEY_REPEAT_DELAY = 150;
+
 export default class NgcOmniboxController {
   static get $inject() {
-    return [];
+    return ['$document', '$element', '$timeout'];
+  }
+
+  constructor($document, $element, $timeout) {
+    this.$document = $document;
+    this.$element = $element;
+    this.$timeout = $timeout;
+
+    this.highlightedIndex = -1; // -1 means nothing is highlighted
   }
 
-  constructor() {
+  $postLink() {
+    this.inputEl = this.$element[0].querySelector('input[role="combobox"]');
 
+    this._suggestionElements = this.$document[0].getElementsByName(
```

Comment:
@onlywei ok, I'm generating unique-ish ids now so hopefully they will never collide. Let me know what you think.